lib/livebook_web/live/session_live/fly_runtime_component.ex

```diff
@@ -610,18 +748,40 @@ defmodule LivebookWeb.SessionLive.FlyRuntimeComponent do
         }
   end
 
-  defp specs_changeset(config, attrs \\ %{}) do
-    defaults = %{
+  defp config_secret_names(hub_secrets) do
+    names =
+      for %{name: name} <- hub_secrets,
+          String.starts_with?(name, @config_secret_prefix),
+          do: name
+
+    Enum.sort(names)
+  end
+
+  defp load_config_defaults(socket) do
+    config_defaults = socket.assigns.config_defaults
+
+    socket
+    |> assign(
+      token: config_defaults["token"],
+      app_name: config_defaults["app_name"],
```

Comment:
Curiosity: why token and app name are not part of specs, so they are all handled in a single place? I guess they have separate loading workflows?

Reply:
Yeah, they affect loading and there is a fair amount of logic that uses these values (conditionals, requests), so handling them individually feels cleaner. Also, when they are empty, instead of showing form errors, we show relevant message boxes with instructions.

Initially I was going for a single changeset, but I ended up with this approach, and with the nested volumes form we can't really include everything in the changeset anyway.